src/app/containers/ConsentBanner/Canonical/logic/index.js

```diff
@@ -0,0 +1,82 @@
+import Cookie from 'js-cookie';
+import setCookieOven from './setCookieOven';
+
+const PRIVACY_COOKIE = 'ckns_privacy';
+const EXPLICIT_COOKIE = 'ckns_explicit';
+const POLICY_COOKIE = 'ckns_policy';
+const COOKIE_EXPIRY = 365;
+const BANNER_APPROVED = '1';
+const POLICY_APPROVED = '111';
+const POLICY_DENIED = '000';
+
+const onClient = typeof window !== 'undefined';
+
+const setCookie = (name, value) =>
+  Cookie.set(name, value, { expires: COOKIE_EXPIRY });
+
+const setPolicyCookie = (value, logger) => {
+  setCookie(POLICY_COOKIE, value);
+  setCookieOven(POLICY_COOKIE, value, logger);
+};
+
+const seenPrivacyBanner = () => Cookie.get(PRIVACY_COOKIE) === BANNER_APPROVED;
+const seenCookieBanner = () => Cookie.get(EXPLICIT_COOKIE) === BANNER_APPROVED;
+const policyCookieSet = () => !!Cookie.get(POLICY_COOKIE);
+
+const setSeenPrivacyBanner = () => setCookie(PRIVACY_COOKIE, BANNER_APPROVED);
+const setDefaultPolicy = logger => setPolicyCookie(POLICY_DENIED, logger);
+const setAppovedPolicy = logger => setPolicyCookie(POLICY_APPROVED, logger);
+const setDismissedCookieBanner = () =>
+  setCookie(EXPLICIT_COOKIE, BANNER_APPROVED);
+
+const consentBannerUtilities = ({
+  setShowPrivacyBanner,
+  setShowCookieBanner,
+  logger,
+}) => {
+  const runInitial = () => {
+    if (onClient) {
+      if (!seenPrivacyBanner()) {
+        setShowPrivacyBanner(true);
+        setSeenPrivacyBanner();
+      }
+
+      if (!seenCookieBanner()) {
+        setShowCookieBanner(true);
```

Comment:
To double check - we expect the renderer to show the Privacy banner, and _then_ the Cookie banner, right?

So this function call won't necessarily display the cookie banner right away?

Maybe add a comment to that effect here :)

Reply:
This is stated in the Readme
>NB: It is down to the application to implement the rendering logic. On a fresh visit both setShowPrivacyBanner and setShowCookieBanner functions will be called with true. The privacy banner should be shown first, followed then by the cookie banner.

Reply:
I meant in the code itself.

Reply:
done :)